IdentityCore/src/telemetry/request_telemetry/MSIDLastRequestTelemetry.m

```diff
@@ -248,6 +279,17 @@ - (void)saveTelemetryToDisk
     NSString *saveLocation = [self filePathToSavedTelemetry];
     if (saveLocation)
     {
+        // Some testing has determined that 75 errors corresponds to an archive size of about 8kb. 
+        if (_errorsInfo.count > maxErrorCountToArchive)
+        {
+            MSID_LOG_WITH_CTX_PII(MSIDLogLevelVerbose, nil, @"Telemetry size over limit when saving to disk, cutting down to limit", nil);
+            
+            NSRange rangeToRemove;
+            rangeToRemove.location = 0;
+            rangeToRemove.length = _errorsInfo.count - maxErrorCountToArchive;
+            [_errorsInfo removeObjectsInRange:rangeToRemove];
```

Comment:
any race conditions from multiple threads?

Reply:
Nope, all calls to saveTelemetryToDisk happen within dispatch barriers.